src/iam/iam-base.ts

```diff
@@ -180,7 +180,8 @@ export class IAMBase {
         walletProvider?: WalletProvider;
     }): Promise<AccountInfo | undefined> {
         const { privateKey, rpcUrl } = this._connectionOptions;
-        if (this._executionEnvironment === ExecutionEnvironment.NODE) {
+        const isPrivateKeyInBrowser = this._executionEnvironment === ExecutionEnvironment.BROWSER && privateKey;
```

Comment:
`isPrivateKeyInBrowser` and `!privateKey` contradicts each other
```suggestion
       if (privateKey) { ... signer from private key}
      if (this._executionEnvironment == ExecutionEnvironment.BROWSER) { throw new Error(ERROR_MESSAGES.PRIVATE_KEY_NOT_PROVIDED)
```

Reply:
Ok, so you prefer to create below an if statement that uses the same part of code?

Reply:
I mean that if private key is provided, then it should be used to create signer regardless of any other conditions